include/emds/emds.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @defgroup emds Emergency Data Storage
+ *  @ingroup emds
+ *  @{
+ *  @brief Emergency Data Storage API
+ */
+
+#ifndef EMDS_H__
+#define EMDS_H__
+
+#include <stddef.h>
+#include <sys/types.h>
+#include <sys/util.h>
+#include <sys/slist.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct emds_entry
+ *
+ * Information about entries used to store data in the emergency data storage.
+ */
+struct emds_entry {
+	/** Unique id for each static and dynamic entry. */
+	uint16_t id;
+	/** Pointer to data that shall be stored. */
+	uint8_t *data;
+	/** Length of data that shall be stored. */
+	size_t len;
+}
+
+/**
+ * @struct emds_dynamic_entry
+ *
+ * Entries with a node element, used for dynamic entries. These are registered
+ * using a call to emds_entry_add().
+ */
+struct emds_dynamic_entry {
+	struct emds_entry entry;
+	sys_snode_t node;
+}
+
+/**
+ * @brief Define a static entry for emergency data storage items.
+ *
+ * @param name The entry name.
+ * @param _id Unique id for entry. This value and not overlap with any other
+ *            value.
+ * @param _data Data pointer to be stored at emergency data store.
+ * @param _len Length of data to be stored at emergency data store.
+ *
+ * This creates a variable _name prepended by emds_.
+ */
+#define EMDS_STATIC_ENTRY_DEFINE(_name, _id, _data, _len)                      \
+	(const STRUCT_SECTION_ITERABLE(emds_entry, emds_##_name) = {           \
```

Comment:
I added the () because if not we do not do that, we get from the checkpatch.pl script: WARNING: Macros with complex values should be enclosed in parentheses #11: FILE: include/emds/emds.h:62:

Reply:
I think same macros don't fail in zephyr because this warning is ignored: https://github.com/zephyrproject-rtos/zephyr/blob/97318703e47fca6842029ace3c310f7562d92e5a/.checkpatch.conf#L28

Reply:
It is not ignored in sdk-nrf. Trond had to emphasize the issue before merging if it is met
https://github.com/nrfconnect/sdk-nrf/pull/3753#issuecomment-776683681